command_namedport.go

```diff
@@ -0,0 +1,121 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os/exec"
+	"strconv"
+	"strings"
+
+	"github.com/urfave/cli"
+)
+
+type namedPort struct {
+	Name string `json:"name"`
+	Port int    `json:"port"`
+}
+
+const (
+	create = 1
+	delete = 2
+)
+
+func cmdCreateNamedPort(c *cli.Context) error {
+	return cmdChangeNamedPort(c, create)
+}
+
+func cmdDeleteNamedPort(c *cli.Context) error {
+	return cmdChangeNamedPort(c, delete)
+}
+
+func cmdChangeNamedPort(c *cli.Context, action int) error {
+	if len(c.Args()) != 2 {
+		return fmt.Errorf("missing command argument, expected [INSTANCE_GROUP] NAME:PORT")
+	}
+	verbose := c.GlobalBool("v")
+	instanceGroup := c.Args()[0]
+	if len(instanceGroup) == 0 {
+		return fmt.Errorf("missing Compute Instance Group command argument")
+	}
+	parts := strings.Split(c.Args()[1], ":")
+	if len(parts) == 0 {
+		return fmt.Errorf("invalid NAME:PORT command argument")
+	}
+	port, err := strconv.Atoi(parts[1])
+	if err != nil {
+		return fmt.Errorf("PORT must be a positive integer:%v", err)
```

Comment:
Afaik, this is incorrect, Atoi supports negative integers (https://play.golang.org/p/3m-NlgsIeS3)

Reply:
yes but ports not